internal/cmd/skupper/connector/kube/connector_generate.go

```diff
@@ -42,12 +38,10 @@ func NewCmdConnectorGenerate() *CmdConnectorGenerate {
 }
 
 func (cmd *CmdConnectorGenerate) NewClient(cobraCommand *cobra.Command, args []string) {
-	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), cobraCommand.Flag("kubeconfig").Value.String())
+	cli, err := client.NewGenerateCmdClient(cobraCommand.Flag("namespace").Value.String())
```

Comment:
This is not necessary, I suggest to just keep
`cmd.namespace = cli.Namespace` in the NewClient method. 

If generate methods are not checking with a client it is fine to just not call them.